ci-operator/step-registry/cucushift/installer/rehearse/aws/cases/all-regions/provision/cucushift-installer-rehearse-aws-cases-all-regions-provision-commands.sh

```diff
@@ -0,0 +1,533 @@
+}
+
+function report_health_check_result()
+{
+  local region=$1
+  local fail_or_pass=$2
+  echo ">>> ${fail_or_pass}: HEALTH CHECK: ${region} $(get_cluster_name $region)"
+  cat <<< "$(jq --arg region ${region} --arg m ${fail_or_pass} '.[$region].health_check_result = $m' "${RESULT}")" > ${RESULT}
+}
+
+
+# -------------------------------------------------------------------------------------
+# generate regions
+# -------------------------------------------------------------------------------------
+echo "Getting regions for test ..."
+generate_regions
+echo "Following regions will be tested:"
+cat ${REGIONS_LIST}
+
+# -------------------------------------------------------------------------------------
+# init result file
+# -------------------------------------------------------------------------------------
+echo "Creating result file ..."
+t=$(mktemp)
+while IFS= read -r region; do
+    cat > ${t} << EOF
+{
+  "region": "${region}",
+  "cluster_name": "$(get_cluster_name "${region}")",
+  "install_dir": "$(get_install_dir "${region}")",
+  "infra_id": "NA",
+  "install_result": "NA",
+  "health_check_result": "NA",
+  "destroy_result": "NA",
+  "metadata": ""
+}
+EOF
+    cat <<< "$(jq  --argjson info "$(<${t})" --arg region $region '. += {($region): $info}' "${RESULT}")" > ${RESULT}
+done < ${REGIONS_LIST}
+
+# -------------------------------------------------------------------------------------
+# Create cluster
+# -------------------------------------------------------------------------------------
+total=$(cat $REGIONS_LIST | wc -l)
+i=0
+while IFS= read -r region; do
+    set +o errexit
+    let i+=1
+    # init result
+    cluster_name=$(get_cluster_name "${region}")
+    install_dir=$(get_install_dir "${region}")
+    mkdir -p ${install_dir}
+
+    echo "================================================================"
+    echo "Creating cluster [${region}][${cluster_name}], ${i}/${total}"
+    echo "================================================================"
+    create_install_config $region $cluster_name $install_dir
+
+    # create ignition configs
+    openshift-install create ignition-configs --dir ${install_dir} &
+    wait "$!"
+    install_ret="$?"
+    ret=$((ret+install_ret))
+    if [ $install_ret -ne 0 ]; then
+      echo "Failed ignition configs, saving metadata.json ... "
+      report_install_result "${region}" "FAIL"
+      continue
+    else
+      echo "Created ignition configs, saving metadata.json and infraid ... "
+
+      cp ${install_dir}/metadata.json ${SHARED_DIR}/metadata.${region}.json
+
+      metadata_b64="$(cat ${install_dir}/metadata.json | base64 -w0)"
+      infra_id="$(cat ${install_dir}/metadata.json | jq -r '.infraID')"
+      cat <<< "$(jq --arg metadata ${metadata_b64} --arg region ${region} '.[$region].metadata = $metadata' "${RESULT}")" > ${RESULT}
+      cat <<< "$(jq --arg i ${infra_id} --arg region ${region} '.[$region].infra_id = $i' "${RESULT}")" > ${RESULT}
+    fi
+
```

Comment:
not sure if we want to enable spot instance for this testing, as we'll get about 30 clusters to be created for each test

Reply:
if spot instance is not stable enough, maybe we can run arm arch instances + replica: 2 to save some cost.

Reply:
This is also remind me, in v4-installer-autotest, the all regions testing randomly selected arm arch or amd arch for more coverage, also attached some storage test cases. Do we have plan to migrate them in prow ci ?

Reply:
> arch instances + replica: 2

This reminds me to use single AZ to reduce EIP usage, will update.

> spot instance

Good point, I can add an option to control this behavior, and set it as default, if it's not stable enough, we can switch back.

> arm arch or amd arch

Duplicate jobs for arm and set to f60 for them?

Reply:
> Duplicate jobs for arm and set to f60 for them?

If I remember right, not each region has arm arch instances.

Reply:
ARM instances are available in all regions

Reply:
If that, I am okay with either ways. "Duplicate jobs for arm and set to f60 for them" or "randomly select instance arch"

Reply:
> also attached some storage test cases.

My fault, that case is for all instance types coverage.

Reply:
is the all region test just triggered once every release or controlled by job frenquency(f60)? if we just trigger once all region test every release, randomly pick architecture for all region will save cost becasue this will iterate all region once, if that we only need to use multiarch payload and give arch info for func `create_install_config`.

Reply:
@lwan-wanglin  We use two worker nodes and spot instances for both worker and master nodes, I think the cost is acceptable, and this will give us more confidence in the report. Of course, we can adjust anytime if any issues come up. WDYT?

Reply:
if that, I am ok too